imap_processing/ultra/l1b/cullingmask.py

```diff
@@ -25,14 +25,32 @@ def calculate_cullingmask(
     cullingmask_dataset : xarray.Dataset
         Dataset containing the data.
     """
-    cullingmask_dict = {}
-    cullingmask_dict["spin_number"] = extendedspin_dataset["spin_number"]
-    cullingmask_dict["energy_bin_geometric_mean"] = extendedspin_dataset[
-        "energy_bin_geometric_mean"
-    ]
-
-    # TODO: add more data to cullingmask_dict.
-
-    cullingmask_dataset = create_dataset(cullingmask_dict, name, "l1b", data_version)
+    # If the spin rate was too high or low then the spin should be thrown out.
+    rate_mask = (
+        extendedspin_dataset["quality_attitude"] & ImapAttitudeUltraFlags.SPINRATE.value
+    ) == 0
+    good_attitude = extendedspin_dataset["spin_number"].values[rate_mask]
+    good_attitude_dataset = extendedspin_dataset.sel(spin_number=good_attitude)
+
+    # Counts in every energy bin should be equal to zero to throw out the spin.
+    zero_counts = (
+        (
+            good_attitude_dataset["quality_ena_rates"]
+            & ImapRatesUltraFlags.ZEROCOUNTS.value
+        )
+        != 0
+    ).all(dim="energy_bin_geometric_mean")
+    zero_counts_mask = ~zero_counts
+    non_zero_dataset = good_attitude_dataset.sel(
+        spin_number=good_attitude_dataset["spin_number"][zero_counts_mask]
+    )
```

Comment:
Can these masks be combined for a large boolean "where" condition? It might also help to not compare to `== 0` and `!= 0` separately and let the "mask" nots work themselves out. It is tough to follow the logic between the cases for me currently because the opposite negations.

```
condition = ((ds.quality_ena_rates & ImapRatesUltraFlags.ZEROCOUNTS.value == 0) | (ds.quality_attitude & ImapAttitudeUltraFlags.SPINRATE.value))
```